lib/SIL/Utils/OptimizationRemark.cpp

```diff
@@ -180,31 +184,35 @@ static SourceLoc inferOptRemarkSearchBackwards(SILInstruction &i) {
 
 SourceLoc swift::OptRemark::inferOptRemarkSourceLoc(
     SILInstruction &i, SourceLocInferenceBehavior inferBehavior) {
-  auto loc = i.getLoc().getSourceLoc();
-
-  // Do a quick check if we already have a valid loc. In such a case, just
-  // return. Otherwise, we try to infer using one of our heuristics below.
-  if (loc.isValid())
-    return loc;
+  // Do a quick check if we already have a valid loc and it isnt an inline
+  // loc. In such a case, just return. Otherwise, we try to infer using one of
+  // our heuristics below.
```

Comment:
I'm not sure this comment makes sense any more. Now it "just returns" only if there's no inline scope.

Also, can you comment on why there's a check for debug scope here? I don't get it.

Reply:
The check for the debug scope is to ensure that remarks are always associated with the outermost scope. I get that now. I think the comment wording is slightly off now with this change.

Reply:
The check for the debug scope is in case this runs against SIL without debug scopes. I'll do an update.